packages/react-component-library/src/components/DatePicker/DatePicker.tsx

```diff
@@ -378,7 +359,7 @@ export const DatePicker = ({
                 variant={BUTTON_VARIANT.TERTIARY}
                 size={COMPONENT_SIZE.SMALL}
                 onClick={() => {
-                  if (isDateDisabled(today, disabledDays)) {
+                  if (isDateInMatcher(today, disabledDays)) {
```

Comment:
Hmm might make sense to do `calculateDateValidity(today, disabledDays) !== DATE_VALIDITY.VALID` instead (not that it's likely that `today` is an invalid date, but keeps `calculateDateValidity` as a single way of working out if a date is valid).